Build issue: implicit instantiation of undefined template in `ComponentDataJSONParser.hpp`

### Build error message(s)

When building GridKit with clang 16 a build error related to implicit instantiation of string stream occurs:
```console
In file included from /.../src/gridkit/GridKit/examples/PhasorDynamics/Tiny/TwoBus/Basic/TwoBusBasicJson.cpp:19:
In file included from /.../src/gridkit/GridKit/src/Model/PhasorDynamics/SystemModelDataJSONParser.hpp:6:
/.../src/gridkit/GridKit/src/Model/PhasorDynamics/ComponentDataJSONParser.hpp:67:29: error: implicit instantiation of undefined template 'std::basic_stringstream<char>'
          std::stringstream msg;
                            ^
/opt/local/libexec/llvm-16/bin/../include/c++/v1/iosfwd:134:32: note: template is declared here
    class _LIBCPP_TEMPLATE_VIS basic_stringstream;
```


### GridKit™ version

`develop` at commit [7feb6842](https://github.com/ORNL/GridKit/commit/7feb6842daaefa792d4cd23a1d0bcd090860bd20)

### System and environment details

MacOS with clang 16.0.6

### Additional information

This build error does not occur when building with GCC.

--- a/src/Model/PhasorDynamics/Bus/BusDataJSONParser.hpp
+++ b/src/Model/PhasorDynamics/Bus/BusDataJSONParser.hpp
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <sstream>
 #include <stdexcept>
 
 #include <Model/PhasorDynamics/Bus/BusData.hpp>

--- a/src/Model/PhasorDynamics/ComponentDataJSONParser.hpp
+++ b/src/Model/PhasorDynamics/ComponentDataJSONParser.hpp
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <sstream>
 #include <stdexcept>
 
 #include <Model/PhasorDynamics/ComponentData.hpp>

--- a/src/Model/PhasorDynamics/SystemModelDataJSONParser.hpp
+++ b/src/Model/PhasorDynamics/SystemModelDataJSONParser.hpp
@@ -1,5 +1,6 @@
 #pragma once
 
+#include <sstream>
 #include <stdexcept>
 
 #include <Model/PhasorDynamics/Bus/BusDataJSONParser.hpp>

--- a/tests/UnitTests/PhasorDynamics/GovernorTgov1Tests.hpp
+++ b/tests/UnitTests/PhasorDynamics/GovernorTgov1Tests.hpp
@@ -63,9 +63,9 @@ namespace GridKit
       {
         TestStatus success = true;
 
-        using BusType          = PhasorDynamics::BusData<ScalarT, IdxT>::BusType;
-        using GenrouParameters = PhasorDynamics::GenrouData<ScalarT, IdxT>::Parameters;
-        using GenrouPorts      = PhasorDynamics::GenrouData<ScalarT, IdxT>::Ports;
+        using BusType          = typename PhasorDynamics::BusData<ScalarT, IdxT>::BusType;
+        using GenrouParameters = typename PhasorDynamics::GenrouData<ScalarT, IdxT>::Parameters;
+        using GenrouPorts      = typename PhasorDynamics::GenrouData<ScalarT, IdxT>::Ports;
 
         PhasorDynamics::BusData<ScalarT, IdxT> busdata;
         busdata.bus_id   = 0;

--- a/tests/UnitTests/Utilities/CaseFormatTests.hpp
+++ b/tests/UnitTests/Utilities/CaseFormatTests.hpp
@@ -28,7 +28,7 @@ namespace GridKit
       {
         using namespace GridKit::PhasorDynamics;
         using BusData = BusData<RealT, IdxT>;
-        using BusType = BusData::BusType;
+        using BusType = typename BusData::BusType;
 
         const char data[] =
             R"({
@@ -133,7 +133,7 @@ namespace GridKit
       {
         using namespace GridKit::PhasorDynamics;
         using BusData = BusData<RealT, IdxT>;
-        using BusType = BusData::BusType;
+        using BusType = typename BusData::BusType;
 
         const char data[] =
             R"({

